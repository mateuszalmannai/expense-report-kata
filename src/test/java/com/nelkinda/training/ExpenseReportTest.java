package com.nelkinda.training;

import org.approvaltests.ApprovalUtilities;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.util.Collections;
import java.util.Date;

public class ExpenseReportTest {

  @Test
  void emptyReportTest() {
    ByteArrayOutputStream output = new ApprovalUtilities().writeSystemOutToStringBuffer();
    ExpenseReport report = new ExpenseReport();
    report.printReport(Collections.emptyList(), new Date(0));
    Approvals.verify(output);
  }
}
