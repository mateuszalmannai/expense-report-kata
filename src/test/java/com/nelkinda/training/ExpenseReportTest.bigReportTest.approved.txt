Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	5000	 
Dinner	5001	X
Breakfast	1000	 
Breakfast	1001	X
Car Rental	50	 
Car Rental	2147483647	 
Meal expenses: 12002
Total expenses: -2147471597
